src/main/java/com/example/restapp/dao/AbstractDAO.java

```diff
@@ -4,13 +4,13 @@
 
 public abstract class AbstractDAO {
 
-	protected BeanSession session;
-
-	public void setBeanSession(BeanSession session) {
+	private final BeanSession session;
```

Comment:
That's OO purity. Probably overkill. I don't mind making it protected, but the correct encapsulation approach is for the subclass to save the reference to session as a private member if it does not want to keep calling `getBeanSession()` / `beanSession()` / `session()` everywhere. I think a protected final variable is totally fine. Too much OO purity is overkill. I don't mind changing it back to protected. Thoughts?